docs/js/sponsors.js

```diff
@@ -0,0 +1,481 @@
+                    ${cards.join('')}
+                </div>
+            </div>
+        </div>
+    `;
+}
+
+/**
+ * スポンサーカードを作成
+ * @param {Object} sponsor - スポンサー情報
+ * @param {string} type - スポンサータイプ
+ * @returns {string|null} カードHTML
+ */
+function createSponsorCard(sponsor, type) {
+    const name = sponsor.name || '企業名未設定';
+    const logoUrl = sponsor.avatar || sponsor.logo || '';
+    const sponsorId = sponsor.id || generateRandomId();
+    
+    // ロゴがない場合は表示しない
+    if (!logoUrl) {
+        return null;
+    }
+
+    // カードのサイズはCSSクラスで制御
+    
+    return `
+        <div class="sponsor-card sponsor-card-${type}" data-sponsor-id="${sponsorId}">
+            <div class="sponsor-logo-wrapper overflow-container">
+                <img src="${logoUrl}" alt="${name}" loading="lazy" 
+                     class="image-border-fix sponsor-logo-img"
+                     data-sponsor-name="${name.replace(/"/g, '&quot;')}"
+                     data-sponsor-type="${type}">
+            </div>
+        </div>
+    `;
+}
+
+/**
+ * ランダムIDを生成
+ * @returns {string} ランダムID
+ */
+function generateRandomId() {
+    return Math.random().toString(36).substr(2, 9);
+}
+
+// ==========================================
+// モーダル関連
+// ==========================================
+
+/**
+ * スポンサーモーダルを開く
+ * @param {string} sponsorId - スポンサーID
+ */
+function openSponsorModal(sponsorId) {
+    const sponsor = sponsorsData.find(s => s.id === sponsorId);
+    if (!sponsor) return;
+    
+    const modal = document.getElementById('sponsorModal');
+    const modalLogo = document.getElementById('modalLogo');
+    const modalInfo = document.getElementById('modalInfo');
+    
+    // ロゴの表示
+    renderModalLogo(modalLogo, sponsor);
+    
+    // 情報の表示
+    renderModalInfo(modalInfo, sponsor);
+    
+    // モーダルを表示
+    modal.classList.add('active');
+    document.body.style.overflow = 'hidden';
+}
+
+/**
+ * モーダルのロゴをレンダリング
+ * @param {HTMLElement} element - ロゴ表示要素
+ * @param {Object} sponsor - スポンサー情報
+ */
+function renderModalLogo(element, sponsor) {
+    const logoUrl = sponsor.avatar || sponsor.logo || '';
+    const logoHtml = logoUrl
+        ? `<img src="${logoUrl}" alt="${sponsor.name}" class="image-border-fix">`
+        : `<div class="modal-logo-fallback">${sponsor.name}</div>`;
+    
+    element.innerHTML = `
+        <div class="modal-logo-inner overflow-container">
+            ${logoHtml}
+        </div>
+    `;
+}
+
+/**
+ * モーダルの情報をレンダリング
+ * @param {HTMLElement} element - 情報表示要素
+ * @param {Object} sponsor - スポンサー情報
+ */
+function renderModalInfo(element, sponsor) {
+    const links = getSponsorLinks(sponsor);
+    
+    element.innerHTML = `
+        <div style="display: flex; flex-direction: column; height: 100%; min-height: 300px;">
+            <div style="flex: 1;">
+                <h4 style="margin-bottom: 1.5rem;">
+                    ${sponsor.name}
+                </h4>
+                ${sponsor.pr && sponsor.pr !== '-' ? `
```

Comment:
ここでこんな感じのロジックにして表示制御してみるのはどうでしょう？
```suggestion
                ${sponsor.pr.length >= CONFIG.CONTENT.MIN_PR_LENGTH ? `
```

Reply:
ありがとうございます、修正しました！